### PR TITLE
change rocksdb init: use OptionsUtil

### PR DESCRIPTION
**Pull request by StevenLuMT**
Descriptions of the changes in this PR:

### Motivation

1. some old parameters in rocksDB is not configurable
2. for all the tuning of rocksdb in the future, there is no need to update the code or introduce configuration to bookie

### Changes

1)   rocks all old parameter change to be configurable
2)  use OptionsUtil to init all params for rocksdb

the old pr #3006  has some rebase error,open a new pr

### CODE REVIEW

**Comment by StevenLuMT** [comment]
rerun failure checks

**Comment by StevenLuMT** [comment]
rerun failure checks

**Comment by StevenLuMT** [comment]
rerun failure checks

**Comment by StevenLuMT** [comment]
@eolivelli @pkumar-singh @zymap
If you have time, please help me review it, thank you

**Comment by StevenLuMT** [comment]
@eolivelli @pkumar-singh @zymap @mauricebarnum 
If you have time, please help me review it, thank you

**Comment by StevenLuMT** [comment]
rerun failure checks

**Comment by StevenLuMT** [comment]
@eolivelli @pkumar-singh @zymap @mauricebarnum
If you have time, please help me review it, thank you

**Comment by StevenLuMT** [comment]
@eolivelli @pkumar-singh @zymap @mauricebarnum
If you have time, please help me review it, thank you

**Comment by StevenLuMT** [comment]
@eolivelli @pkumar-singh @zymap @mauricebarnum @hangc0276 
If you have time, please help me review it, thank you

**Comment by StevenLuMT** [comment]
@eolivelli @pkumar-singh @zymap @mauricebarnum @hangc0276
If you have time, please help me review it, thank you

**Comment by StevenLuMT** [comment]
@dlg99 @eolivelli @pkumar-singh @zymap @mauricebarnum @hangc0276
If you have time, please help me review it, thank you

**Comment by StevenLuMT** [comment]
@dlg99 @eolivelli @pkumar-singh @zymap @mauricebarnum @hangc0276
If you have time, please help me review it, thank you

**Comment by StevenLuMT** [comment]
@dlg99 @eolivelli @pkumar-singh @zymap @mauricebarnum @hangc0276
If you have time, please help me review it, thank you

**Comment by StevenLuMT** [comment]
@dlg99 @eolivelli @pkumar-singh @zymap @mauricebarnum @hangc0276
If you have time, please help me review it, thank you

**Comment by StevenLuMT** [comment]
@dlg99 @eolivelli @pkumar-singh @zymap @mauricebarnum @hangc0276 @sijie 
If you have time, please help me review it, thank you

**Comment by StevenLuMT** [comment]
@dlg99 @eolivelli @pkumar-singh @zymap @hangc0276 @lordcheng10
If you have time, please help me review it, thank you.

**Comment by lordcheng10** [comment]
LGTM

**Comment by lordcheng10** [comment]
LGTM

**Review by dlg99 — APPROVED** [review]
LGTM

**Review by hangc0276 — COMMENTED** [review]
Good job.

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/ServerConfiguration.java [hunk 1]
```diff
@@ -3777,4 +3787,76 @@ public ServerConfiguration setDataIntegrityStampMissingCookiesEnabled(boolean en
...
+     * Get default rocksdb conf.
+     *
+     * @return String configured default rocksdb conf.
+     */
+    public String getDefaultRocksDBConf() {
+        String defaultPath = "conf/default_rocksdb.conf";
+        URL defURL = getClass().getClassLoader().getResource(defaultPath);
+        if (defURL != null) {
+            defaultPath = defURL.getPath();
+        }
```
Comment (lhotari) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/ServerConfiguration.java:3801]: This is very surprising behavior that suddenly the classloader is used to locate the resources.
At least there should be a check that the scheme of the URL is a file which isn't embedded in a Jar file.

**Pull request merged**